### PR TITLE
HTML: Fix inconsistent formatting of self-closing tags

### PR DESCRIPTION
**Pull request by fisker**
- [stable playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAzgBwIZQHyoCWUmArjAAQwCemcAvADrgAWcYA1gEYQAezAejwVUArLjyMoGHPhkSiJclVoNmYNpx78QQ0eLljZk6QYLEylGnSat23PoIJGJIADQgImGIWjpkoNgATkEQAO4ACsEI-ijYAG4QhAAm7iBcQdiccDAAyjhgxADmyDBBpHAeLDAAtgA2AOoshPDiYHC5MS2E8S3UyODo-h7E6HBBMBGZRTXYyABm2HVjHgBW6LwAQpnZedg1cAAyxHALSysg67y5xXVwAIqkEPBny5UgOEFjQQPV9WmYILEGANFIwFjIAAcAAYPICIGMGplMANAXBvvFTh4AI5PeBTLyxEDYdAAWigcDgySpaSCcFxhDpU2wMzmSEWbw8YxqhFK5Xe6FuDzxp3Z53eMGwXFByXByAATB4ythCHVigBhCA1WYDKDQLEgUhjAAqUtiHIu8QqAEkoNTYLkwEDvABBO25Gh3V5jAC+PqAA)

- [PR playground](https://deploy-preview-7395--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAzgBwIZQHyoCWUmArjAAQwCemcAvADrgAWcYA1gEYQAezAejwVUArLjyMoGHPhkSiJclVoNmYNpx78QQ0eLljZk6QYLEylGnSat23PoIJGJIADQgImGIWjpkoNgATkEQAO4ACsEI-ijYAG4QhAAm7iBcQdiccDAAyjhgxADmyDBBpHAeLDAAtgA2AOoshPDiYHC5MS2E8S3UyODo-h7E6HBBMBGZRTXYyABm2HVjHgBW6LwAQpnZedg1cAAyxHALSysg67y5xXVwAIqkEPBny5UgOEFjQQPV9WmYILEGANFIwFjIAAcAAYPICIGMGplMANAXBvvFTh4AI5PeBTLyxEDYdAAWigcDgySpaSCcFxhDpU2wMzmSEWbw8YxqhFK5Xe6FuDzxp3Z53eMGwXFByXByAATB4ythCHVigBhCA1WYDKDQLEgUhjAAqUtiHIu8QqAEkoNTYLkwEDvABBO25Gh3V5jAC+PqAA)

Fixes: #7038

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/html_tags/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -2552,6 +2569,22 @@ printWidth: 1
...
+<!-- self-closing -->
+<span
+  ><input
+    type="checkbox"
+  />
+</span>
+<span
+  ><span
+    ><input
+      type="checkbox" /></span
```
Comment (fisker) [inline tests/html_tags/__snapshots__/jsfmt.spec.js.snap]: Seems all good except this one.

Comment (fisker) [reply tests/html_tags/__snapshots__/jsfmt.spec.js.snap]: Seems breakable [playground](https://deploy-preview-7395--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAzgBwIZQHwY76oCWUmArjAAQwCemcAvADrgAWcYA1gEYQAPNrQbM2YTj35CQIxqw5c+g4fXnjJymXLGKpK2Wt0Sl01aIUn92o5c1nDFjaYM67Lm071bz67w4B6AgCsXGDQ-BAAGhAITBgSaHRkUGwAJzSIAHcABXSEZJRsADcIEgATaJBeNOweOBgAZRwwMgBzZBg0ijgY9hgAWwAbAHV2EnhQsDhGgomSYom6ZHB0ZJiydDg0mBzatoHsZAAzbCGtmIArdAEAIVr6puwBuAAZMjgTs4uQa4FG9pDOAARQoEHgX3OvRAODSWzSK36wyqmDSZBgIwqMHYyAAHAAGGKoiBbEa1TArVFweHFT4xACOYPgezihRA2HQAFooHA4OU+VU0nBGSQhXtsAcjkhTlCYlsBiROt1oehASCmZ9pd9oTBsLxMeVscgAEwxLrYEhDdoAYQgA0OKyg0DpIAoWwAKnrCjKfsUegBJKD82CNMBo+IAQSDjXoQMhWwAvgmgA), should be good

**Comment by fisker** [comment]
Ready for review

File: tests/html_tags/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (evilebottnawi) [reply tests/html_tags/__snapshots__/jsfmt.spec.js.snap]: I think we should not break here, but it is the other problem and we can open issue about it and solve in an other PR

Comment (fisker) [reply tests/html_tags/__snapshots__/jsfmt.spec.js.snap]: no, this need break. Its `printWidth=1` test

Comment (evilebottnawi) [reply tests/html_tags/__snapshots__/jsfmt.spec.js.snap]: Oh yes :+1:

File: src/language-html/printer-html.js [hunk 2]
```diff
@@ -671,7 +671,9 @@ function printOpeningTag(path, options, print) {
            */
           (node.isSelfClosing &&
             needsToBorrowLastChildClosingTagEndMarker(node.parent))
-            ? ""
+            ? node.isSelfClosing
+              ? " "
+              : ""
```
Comment (thorn0) [inline src/language-html/printer-html.js]: `node.isSelfClosing` seems to always be truthy here, see the line 672.

Comment (fisker) [reply src/language-html/printer-html.js]: Another branch above L664

**Pull request merged**